src/lib.rs

```diff
@@ -0,0 +1,1210 @@
+    ///
+    /// Note that this function is same as extend except that it is specialized to work with slices
+    /// instead. If and when Rust gets specialization this function will likely be deprecated (but
+    /// still available).
+    ///
+    /// # Examples
+    /// ```
+    /// #[macro_use] extern crate vob;
+    /// fn main() {
+    ///     let mut v = vob![true];
+    ///     v.extend_from_slice(&vec![false, true]);
+    ///     assert_eq!(v, vob![true, false, true]);
+    /// }
+    pub fn extend_from_slice(&mut self, other: &[bool]) {
+        for &b in other.iter() {
+            self.push(b);
+        }
+    }
+
+    /// Clears the Vob, removing all values.
+    ///
+    /// Note that this method has no effect on the allocated capacity of the Vob.
+    pub fn clear(&mut self) {
+        self.len = 0;
+        self.vec.clear();
+    }
+
+    /// Sets all bits in the Vob to `value`. Notice that this does not change the number of bits
+    /// stored in the Vob.
+    ///
+    /// # Examples
+    /// ```
+    /// #[macro_use] extern crate vob;
+    /// fn main() {
+    ///     let mut v = vob![true, false, true];
+    ///     v.set_all(false);
+    ///     assert_eq!(v, vob![false, false, false]);
+    /// }
+    /// ```
+    pub fn set_all(&mut self, value: bool) {
+        for b in self.vec.iter_mut() {
+            *b = if value { T::max_value() } else { T::zero() };
+        }
+        self.mask_last_block();
+    }
+
+    /// Negates all bits in the Vob.
+    ///
+    /// # Examples
+    /// ```
+    /// #[macro_use] extern crate vob;
+    /// fn main() {
+    ///     let mut v = vob![true, false];
+    ///     v.negate();
+    ///     assert_eq!(v, vob![false, true]);
+    /// }
+    /// ```
+    pub fn negate(&mut self) {
+        for b in self.vec.iter_mut() {
+            *b = !*b;
+        }
+        self.mask_last_block();
+    }
+
+    /// For each bit in this Vob, `and` it with the corresponding bit in `other`, returning `true`
+    /// if this led to any changes or `false` otherwise. The two Vobs must have the same number of
+    /// bits.
+    ///
+    /// # Panics
+    ///
+    /// If the two Vobs are of different length.
+    ///
+    /// # Examples
+    /// ```
+    /// #[macro_use] extern crate vob;
+    /// fn main() {
+    ///     let mut v1 = vob![true, false, false];
+    ///     let v2 = vob![true, true, false];
+    ///     v1.and(&v2);
+    ///     assert_eq!(v1, vob![true, false, false]);
+    /// }
+    /// ```
+    pub fn and(&mut self, other: &Vob<T>) -> bool {
+        if self.len != other.len {
+            panic!("Cannot 'and' two Vobs of different length ({}  {})", self.len, other.len);
+        }
+        let mut chngd = false;
+        for (sb, ob) in self.vec.iter_mut().zip(other.vec.iter()) {
+            let ov = *sb;
+            let nv = ov & *ob;
+            if ov != nv {
+                chngd = true;
+            }
+            *sb = nv;
+        }
+        self.mask_last_block();
```

Comment:
If the Vobs are the same length, they will both have the same number of `0`s at the top, and `0 & 0 == 0` so I think you can kill the masking? And in the `or` function too?

Reply:
Good point.